job-service-db/src/main/resources/procedures/processDependentJobs.sql

```diff
@@ -33,68 +33,56 @@ RETURNS TABLE(
 LANGUAGE plpgsql
 AS $$
 BEGIN
-    -- Get a list of jobs that can be processed without delay
-    CREATE TEMPORARY TABLE tmp_dependent_jobs_ready_to_run
+    -- Get a list of jobs that depend on in_job_id
+    CREATE TEMPORARY TABLE tmp_dependent_jobs
         ON COMMIT DROP
     AS
-        SELECT jtd.job_id, jtd.task_classifier, jtd.task_api_version, jtd.task_data, jtd.task_pipe, jtd.target_pipe
-        FROM job_task_data jtd
-        INNER JOIN job j
-            ON j.job_id = jtd.job_id
-        INNER JOIN job_dependency jd
-            ON jd.job_id = jtd.job_id
-        WHERE j.delay = 0
-            AND jd.dependent_job_id = in_job_id
-            AND NOT EXISTS(
-                SELECT NULL
-                FROM job_dependency jd2
-                INNER JOIN job j2
-                    ON j2.job_Id = jd2.dependent_job_id
-                WHERE jd2.job_id = jd.job_id
-                    AND jd2.dependent_job_id <> in_job_id
-                    AND j2.status <> 'Completed');
+        SELECT j.job_id, j.delay
+		FROM job_dependency jd
+		LEFT JOIN job j ON j.job_id = jd.job_id
+		WHERE jd.dependent_job_id = in_job_id;
 
-    RETURN QUERY
-    SELECT * FROM tmp_dependent_jobs_ready_to_run;
+    PERFORM NULL FROM public.job WHERE job.job_id IN (SELECT tmp_dependent_jobs.job_id FROM tmp_dependent_jobs) ORDER BY job.job_id ASC FOR UPDATE;
```

Comment:
- If this lock is needed then I think it should be in the first statement.
- However I can't quite think why it is needed. (I'll think about it more shortly.)

Reply:
As discussed, this can deadlock with a new job at the minute.

Reply:
Commented out this statement to allow merges to proceed.

Reply:
I've raised [SCMOD-6620](https://portal.digitalsafe.net/browse/SCMOD-6620) for this.